### ISSUE TITLE
On the fly SSL registration and renewal with Let's Encrypt.

### ISSUE THREAD

**Issue by deathg0d** [traefik/traefik#7452]
### Do you want to request a *feature* or report a *bug*?

Feature

### What did you expect to see?

Part 1:
I have a server that accepts multiple domains. I am currently using nginx to do the following:
- Forward all requests from my main domain `example.com` to port 3000
- Forward all requests from all other domains (`!example.com`) to port 8080 (the server does not have the knowledge of the complete list of domains i.e. domains that are not `example.com` are simply forwarded to port 8080)

Part 2:
https://github.com/auto-ssl/lua-resty-auto-ssl this plugin further extends nginx to provide ssl certificates to all domains on the fly. That is, the first time you visit a domain on this server, it creates a certificate. For the next requests, these certificates are used (also supports automatic renewal).

To summarize, the system does not have information of any of the domains. As soon as the first request comes in, it creates a ssl certificate on the fly.

Is such a system possible with Traefik? I remember reading about this possibility long time ago but now I am unable to find this in the documentation.

Currently I am using docker to setup Traefik but all the examples I have read online require that I specify Host, for reverse proxy and for ssl generation.

**Comment by jbdoumenjou** [comment]
Hi @deathg0d , 

Thanks for your interest in Traefik ! 

If you talk about ACME [onDemand](https://doc.traefik.io/traefik/v1.7/configuration/acme/#ondemand-deprecated) option in v1.7, it has been deprecated and there is no plan to re-introduce it.

For the actual usage, feel free to check the [migration guide](https://doc.traefik.io/traefik/v2.3/migration/v1-to-v2/#acme-letsencrypt), and the [let's encrypt](https://doc.traefik.io/traefik/v2.3/https/acme/) part of documentation.

For other questions, please come on the [community forum](https://community.traefik.io).
